File: solutions/java/src/taskmanagementsystem/ActivityLog.java
package taskmanagementsystem;

import java.util.Date;
import java.util.UUID;

public class ActivityLog {
    private final String logId;
    private final String action;
    private final User performedBy;
    private final Date timestamp;

    public ActivityLog(String action, User performedBy) {
        this.logId = UUID.randomUUID().toString();
        this.action = action;
        this.performedBy = performedBy;
        this.timestamp = new Date();
    }
}

File: solutions/typescript/src/TaskManagement/ActivityLog.ts
import User from "./User";
import { v4 as uuidv4 } from "uuid";

export default class ActivityLog {
    private logId: string;
    private action: string;
    private performedBy: User;
    private timestamp: Date;

    constructor(action: string, performedBy: User) {
        this.logId = uuidv4();
        this.action = action;
        this.performedBy = performedBy;
        this.timestamp = new Date();
    }
}
